funsor/integrate.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import absolute_import, division, print_function
+
+from collections import OrderedDict
+from six import integer_types
+
+import torch
+
+import funsor.ops as ops
+from funsor.distributions import Gaussian, Delta
+from funsor.interpreter import interpretation, reinterpret
+from funsor.optimizer import Finitary, optimize
+from funsor.terms import Funsor, Number, Reduce, Variable, eager, reflect
+from funsor.torch import Tensor
+
+
+GROUND_TERMS = (Tensor, Gaussian, Delta, Number, Variable)
```

Comment:
We will want to either include `Joint` here, or split `Joint` into a `Finitary(ops.add, ...)` whose terms are ground.